src/Microsoft.Data.SqlClient/netcore/src/Microsoft/Data/SqlClient/TdsParserStateObject.netcore.cs

```diff
@@ -322,14 +322,22 @@ private void ReadSniError(TdsParserStateObject stateObj, uint error)
                             stateObj.SendAttention(mustTakeWriteLock: true);
 
                             PacketHandle syncReadPacket = default;
+                            bool readFromNetwork = true;
                             RuntimeHelpers.PrepareConstrainedRegions();
                             bool shouldDecrement = false;
                             try
                             {
                                 Interlocked.Increment(ref _readingCount);
                                 shouldDecrement = true;
-
-                                syncReadPacket = ReadSyncOverAsync(stateObj.GetTimeoutRemaining(), out error);
+                                readFromNetwork = !PartialPacketContainsCompletePacket();
```

Comment:
The function `PartialPacketContainsCompletePacket` was originally called `IsCompletePacket` and calling it that caused me to miss some problem cases. The partial packet can contain a complete packet as well as fragments or complete other packets (see tests). The current name is very long and very specific but it needs to be called this or something that expresses the same set of information so that it's very clear what the result means. 
If anyone can think of a shorter name that expresses the same thing I'll happily rename it.